deepmd/pt/train/training.py

```
@@ -1064,48 +1077,14 @@ def save_model(self, save_path, lr=0.0, step=0) -> None:
-                    batch_data = next(iter(self.training_data[task_key]))
-            else:
-                if self.validation_data[task_key] is None:
-                    return {}, {}, {}
-                try:
-                    batch_data = next(iter(self.validation_data[task_key]))
-                except StopIteration:
-                    self.validation_data[task_key] = BufferedIterator(
-                        iter(self.validation_dataloader[task_key])
-                    )
-                    batch_data = next(iter(self.validation_data[task_key]))
+            iterator = self.validation_data
```

Ruff (0.8.2)

Use ternary operator `iterator = self.training_data if is_train else self.validation_data` instead of `if`-`else`-block

Replace `if`-`else`-block with `iterator = self.training_data if is_train else self.validation_data`

(SIM108)